src/App.tsx

```diff
@@ -31,30 +37,97 @@ export const App = (): JSX.Element => {
     // eslint-disable-next-line
   }, []);
 
-  useEffect(() => {
-    (async () => {
-      if (useProxy) {
-        await Command.sidecar('grpcproxy', '--log-to-file').spawn();
-        setProxyIsRunning(true);
-      }
-    })();
-    // eslint-disable-next-line
-  }, []);
+  const startProxy = async () => {
+    const command = Command.sidecar('grpcproxy');
+    command.on('close', (data) => {
+      console.log(`grpcproxy command finished with code ${data.code} and signal ${data.signal}`);
+    });
+    command.on('error', (error) => console.error(`grpcproxy command error: "${error}"`));
+    command.stdout.on('data', (line) => console.log(`grpcproxy command stdout: "${line}"`));
+    command.stderr.on('data', (line) => console.log(`grpcproxy command stderr: "${line}"`));
+    const child = await command.spawn();
+    console.log('Proxy pid:', child.pid);
+  };
 
-  useEffect(() => {
-    (async () => {
-      if (useProxy && tdexdConnectUrl && proxyIsRunning) {
-        dispatch(connectProxy());
+  const startAndConnectToProxy = useCallback(async () => {
+    console.log('startAndConnectToProxy');
+    if (useProxy && proxyHealth !== 'SERVING') {
```

Comment:
better checking the status codes instead of the string description now that they are fixed?